jujugui/static/gui/src/app/store/env/base.js

```diff
@@ -282,7 +282,7 @@ YUI.add('juju-env-base', function(Y) {
       // Consider the user unauthenticated until proven otherwise.
       this.userIsAuthenticated = false;
       this.failedAuthentication = false;
-      const credentials = this.get('user').controller;
+      const credentials = this.get('user').controller || {};
```

Comment:
Ditto. The controller property should always be a proper object or the empty object.